app/helpers/vacancies_helper.rb

```diff
@@ -9,10 +9,10 @@ def humanize_array(items)
     items.map(&:humanize).join(", ")
   end
 
-  def location_formatter(items)
+  def location_formatter(items, location_source: :location, join: " ")
     items.map { |value|
-      "#{value['location']} (#{t('jobs.search.number_of_miles', count: value['radius'].to_i)})"
-    }.join(" ")
+      "".html_safe + "#{value[location_source.to_s]} (#{t('jobs.search.number_of_miles', count: value['radius'].to_i)})"
+    }.join("".html_safe + join).html_safe
```

Comment:
There are a few things here. Firstly, this helper is being used with an array of location-containing models, but it makes an assumption that all models will store it's data in column named `location`. This is not the case for `JobPreferences::Location` where we need to read data from `name` column instead. As a result there was a small issue where locations were displayed without the name and this fixes it.

Additionally, it assumed that we want all locations listed in a single line, separated by space. In that particular case I want one location per line. TBH, I don't know why we have a helper that formats a collection of locations instead of just a single location, but nvm.

`"".html_safe + some_string` is the shortest way of escaping potentially dangerous html injections while respecting its `html_safe?`. Default `Array#join` implementation automatically assumes all strings are safe, so there was a potential security issue in the helper.